kubetest2/pkg/app/cmd.go

```diff
@@ -30,51 +31,70 @@ import (
 )
 
 // Run instantiates and executes the kubetest2 cobra command, returning the result
-func Run(deployerName string, newDeployer types.NewDeployer) error {
-	return NewCommand(deployerName, newDeployer).Execute()
+func Run(deployerName, deployerUsage string, newDeployer types.NewDeployer) error {
+	return NewCommand(deployerName, deployerUsage, newDeployer).Execute()
 }
 
 // NewCommand returns a new cobra.Command for kubetest2
-func NewCommand(deployerName string, newDeployer types.NewDeployer) *cobra.Command {
-	opt := &options{}
+func NewCommand(deployerName, deployerUsage string, newDeployer types.NewDeployer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use: fmt.Sprintf("%s %s", shim.BinaryName, deployerName),
 		// we defer showing usage, so that we can include deployer and test
 		// specific usage in RealMain(...)
 		SilenceUsage:  true,
 		SilenceErrors: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runE(cmd, args, opt, deployerName, newDeployer)
+			return runE(cmd, args, deployerName, deployerUsage, newDeployer)
 		},
 	}
-
 	// we implement custom flag parsing below
 	cmd.DisableFlagParsing = true
-
 	return cmd
 }
 
 // runE implements the custom CLI logic
 func runE(
 	cmd *cobra.Command, args []string,
-	opts *options, deployerName string, newDeployer types.NewDeployer,
+	deployerName, deployerUsage string, newDeployer types.NewDeployer,
 ) error {
-	// show help if there are zero arguments
+	// setup the options struct & flags
+	opts := newOptions(deployerName, deployerUsage)
```

Comment:
flags setup moved into here, which made it cleaner to have a `usage()` func that computes usage with the kubetest2 flags + tester usage etc.